src/agni/server/manager/UserManager.java

```diff
@@ -14,10 +16,113 @@ public UserManager(InfoSender infoSender,
         this.infoSender = infoSender;
         this.userDataGuard = userDataGuard;
     }
-
-	@Override
-	public void userRequest(String ip, byte type, String testMessage) {
-		// TODO Auto-generated method stub
-		
+    private boolean friendHasAccepted(String username, String friend) {
+    	String [] friendList = userDataGuard.getFriends(friend); 
+		if(Arrays.asList(friendList).contains(username)) {
+			return true;
+		}
+		else {
+			return false;
+		}
 	}
+    public enum UserRequestType {
+        JOIN_CHAT((byte)0x00),
+        LEAVE_CHAT((byte)0x01),
+        FRIENDS_LIST((byte)0x02),
+        FRIEND_STATUS((byte)0x03),
+        ADD_FRIEND((byte)0x04),
+        LOGOUT((byte)0x05);
+
+        private final byte bytes;
+        private UserRequestType(byte bytes) {
+            this.bytes = bytes;
+        }
+
+        public final byte bytes() {
+            return bytes;
+        }
+    }
```

Comment:
The enum should be near the top of the class definition, a similar place as the private members.